addons/barcodes/static/src/js/barcode_parser.js

```diff
@@ -10,6 +10,7 @@ var rpc = require('web.rpc');
 var BarcodeParser = Class.extend({
     init: function(attributes) {
         this.nomenclature_id = attributes.nomenclature_id;
+        this.nomenclature = attributes.nomenclature;
```

Comment:
With this change, I wonder if it shouldn't better to return a reject promise in the BarcodeParser `load` if there is no `nomenclature_id` and no `nomenclature`.